tensorboard/plugins/custom_scalar/layout.py

```diff
@@ -0,0 +1,50 @@
+
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Contains logic related to laying out the custom scalars dashboard.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+from tensorboard.plugins.custom_scalar import metadata
+
+def set_layout(writer, scalars_layout):
```

Comment:
Why not name this module `summary.py`, call this function `op` and then use `tf.summary.tensor_summary` to write a serialized proto tensor string?

Reply:
You can set the plugin_name on the summary to something like "custom_scalars" to differentiate it. Then you say `from tensorboard import summary` and `summary.custom_scalars_config(<proto object>)` in the training script.

Reply:
Yeah, the plugin_name is already differentiated if you observe summary.py.

That SG. PTAL. I made `op` and `pb` methods to clarify which one is to be used within a  TensorFlow session.